pages/previewPage.html

```diff
@@ -139,10 +139,21 @@
         var newZl = Math.round(zl + zoomStep) - Math.round(zl % zoomStep);
         return Math.max(10, newZl);
       }
+
       window.addEventListener("DOMContentLoaded", () => {
         init();
       });
 
+      window.addEventListener("wheel", (event) => {
+        if (event.ctrlKey === true) {
+          if (event.deltaY > 0) {
+            zoomMinus();
+          } else {
+            zoomPlus();
+          }
+        }
+      });
```

Comment:
```suggestion
      window.addEventListener("wheel", (event) => {
        if (event.ctrlKey === true) {
          event.preventDefault();
          if (event.deltaY > 0) {
            zoomMinus();
          } else {
            zoomPlus();
          }
        }
      }, { passive: false });
```

to prevent scrolling while zooming

this will also need something like #721 to render with dimensions set so zooming works correctly here

Reply:
I just upgraded d2 to v0.2.2 that changed the svg code and that threw off all my janky zoom code.  I'm working on it.

Reply:
> I just upgraded d2 to v0.2.2 that changed the svg code and that threw off all my janky zoom code. I'm working on it.

ok tested a locally compiled d2 version that sets dimensions for svg and the zooming works with that

Reply:
OK, so any fix I do will break on my next upgrade?

Reply:
If you make a fix now it should still work, you would just probably not use the set dimensions flag in the d2 call.

The idea is adding a flag that you could provide to set the dimensions on the svg, which would work with this zooming, so an option is waiting for that and just adding the flag to the d2 command.